back/app/services/projects_finder_admin_service.rb

```diff
@@ -0,0 +1,144 @@
+class ProjectsFinderAdminService
+  UUID_REGEX = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+
+  # SORTING METHODS
+  def self.sort_recently_viewed(scope, current_user)
+    substring_statement = "substring(path, 'admin/projects/(#{UUID_REGEX})')"
+
+    # I first did this with a group by and max, but Copilot suggested
+    # using a window function instead, which is more efficient
+    recent_pageviews_sql = <<-SQL.squish
+      SELECT
+        #{substring_statement}::UUID AS admin_project_id,
+        impact_tracking_pageviews.created_at AS last_viewed_at,
+        ROW_NUMBER() OVER (
+          PARTITION BY #{substring_statement}
+          ORDER BY impact_tracking_pageviews.created_at DESC
+        ) AS rn
+      FROM impact_tracking_pageviews
+      INNER JOIN impact_tracking_sessions
+        ON impact_tracking_pageviews.session_id = impact_tracking_sessions.id
+      WHERE path LIKE '%admin/projects/%'
+        AND impact_tracking_sessions.user_id = '#{current_user.id}'
```

Comment:
Given the amount of data some platforms might have for `impact_tracking_page_views` you'll definitely want to add indexes on `path` and `user_id` or you'll get performance issues